File: src/domain/entities/insight-card.ts
export type InsightCardType =
  | "EARNINGS"
  | "ORDER"
  | "ANNOUNCEMENT"
  | "DEAL"
  | "CORPORATE_ACTION"
  | "NEWS";

export type InsightCardSentiment = "positive" | "negative" | "neutral" | "mixed";

export type InsightCard = {
  id: string;
  /** null when built without a persisted EngineSignal (e.g., direct from DealEvent) */
  signalId: string | null;
  ticker: string;
  companyName: string;
  cardType: InsightCardType;
  headline: string;
  summary: string;
  sentiment: InsightCardSentiment;
  /** 0–1 float from EngineSignal.confidence */
  confidence: number;
  /** 1–5 star rating derived from impactScore */
  rating: number;
  /** 0–100 from EngineSignal.impactScore (pre-market-adjusted) */
  impactScore: number;
  source: string;
  sourceUrl: string | null;
  pdfUrl: string | null;
  publishedAt: Date;
  createdAt: Date;
};

File: engine/app/domain/entities/insight_card.py
from typing import Literal, Optional
from datetime import datetime
from pydantic import BaseModel

InsightCardType = Literal["EARNINGS", "ORDER", "ANNOUNCEMENT", "DEAL", "CORPORATE_ACTION", "NEWS"]
InsightCardSentiment = Literal["positive", "negative", "neutral", "mixed"]

class InsightCard(BaseModel):
    id: str
    signal_id: Optional[str] = None
    ticker: str
    company_name: str
    card_type: InsightCardType
    headline: str
    summary: str
    sentiment: InsightCardSentiment
    confidence: float
    rating: float
    impact_score: float
    source: str
    source_url: Optional[str] = None
    pdf_url: Optional[str] = None
    published_at: datetime
    created_at: datetime
